code401challenges/linkedLists/src/main/java/linkedLists/LinkedList.java

```diff
@@ -0,0 +1,83 @@
+package linkedLists;
+
+public class LinkedList {
+    Node head; // head of list
+
+    public boolean isEmpty() {
+        return length() == 0;
+    }
+
+    int length() {
+        int length = 0;
+        Node current = head;
+
+        while(current != null){
+            length ++;
+
+            current = current.next;
+        }
+        return length;
+    }
+
+   // Method to insert a new node
+    public LinkedList Insert(LinkedList list, String data)
+    {
+        // Create a new node with given data
+        Node newNode = new Node(data);
+        newNode.next = null;
+
+        // if linked list is empty, new node is head
+        if (list.head == null) {
+            list.head = newNode;

```

Comment:
Instead of passing in a LinkedList as a parameter, you could have used `this` in its place.
Ex: `this.head`